lib/pdf/reader/standard_security_handler.rb

```diff
@@ -87,9 +89,19 @@ def decrypt( buf, ref )
       objKey = @encrypt_key.dup
       (0..2).each { |e| objKey << (ref.id >> e*8 & 0xFF ) }
       (0..1).each { |e| objKey << (ref.gen >> e*8 & 0xFF ) }
-      length = objKey.length < 16 ? objKey.length : 16
-      rc4 = RC4.new( Digest::MD5.digest(objKey)[0,length] )
-      rc4.decrypt(buf)
+      if @cfm == :AESV2
+        objKey << 'sAlT'  # Algorithm 1, b)
+        length = objKey.length < 16 ? objKey.length : 16
+        cipher = OpenSSL::Cipher.new("AES-#{length << 3}-CBC")
+        cipher.decrypt
+        cipher.key = Digest::MD5.digest(objKey)[0,length]
+        cipher.iv = buf[0..15]
+        cipher.update(buf[16..-1]) + cipher.final
+      else
+        length = objKey.length < 16 ? objKey.length : 16
+        rc4 = RC4.new( Digest::MD5.digest(objKey)[0,length] )
+        rc4.decrypt(buf)
+      end
     end
```

Comment:
It looks like the added complexity to this method means the method exceeds the maximum complexity. Are you able to refactor it a bit?

The quickest solution might be to move the two branches (aes/rc4) into private methods. An alternative that would be a bit more work but that I'm very interested in is to have distinct classes that implement the aes and rc4 decryption algorithms and somehow pass them into `StandardSecurityHandler`